### PR TITLE
Adding Magiclysm equipment enchanting recipes & missing enchanted equipments

### PR DESCRIPTION
**Pull request by EigentlicherVogel**
#### Summary

SUMMARY: Mods "Porting equipment enchantment recipes from DDA version of Magiclysm"

#### Purpose of change

The enchanted weapons part of Magiclysm is largely inaccessible as of now in BN because of their incredible rarity: both are, as of now, exclusively from the already very rare black dragon lairs. +2 weapons in particular have an extremely low chance of spawning even from these locations, making it statistically unlikely for any normal playthrough to encounter them.

#### Describe the solution

Adding an enchantment guide that can spawn with other high-end magical recipe books that contains recipes to enchant items to +1 and +2.
Also adds several enchanted tools that had been added to DDA but was not present in BN, that can be accessed via this book.

#### Describe alternatives you've considered

-Adding enchanted weapons to more loot itemgroups: too much fine-grain division between mod versions
-Adding more dungeons that drop enchanted weapons: excessive work, also causing too much fine-grain division between mod versions

#### Testing

- [x] Mod can successfully load
- [x] Game can successfully load with mod
- [x] Book correctly spawn with the advanced magical recipe itemgroup at a low probability
- [x] Player can craft enchanted weapons with the required skills, materials and the book

#### Additional context

Magiclysm in DDA contains a proficiency that have to be grinded prior to making these recipes. To balance the lack of proficiency gating, they now require higher skill levels.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Reminder, there is an online linter for checking this sorta thing: http://dev.narc.ro/cataclysm/format.html

File: data/mods/Magiclysm/items/enchanted_melee.json [hunk 1]
```diff
@@ -755,6 +771,22 @@
     "proportional": { "price": 6, "bashing": 1.2, "cutting": 1.2, "weight": 0.8 },
     "relative": { "to_hit": 2 }
   },
+  {
+    "type": "GENERIC",
+    "id": "lucerne_plus_one",
+    "copy-from": "lucern_hammer",
+    "name": { "str": "lucerne hammer +1", "str_pl": "lucerne hammers +1" },
+    "proportional": { "price": 3.0, "price_postapoc": 3.0, "weight": 0.9, "melee_damage": { "all": 1.1 } },
```
Comment (chaosvolt) [inline data/mods/Magiclysm/items/enchanted_melee.json:779]: `melee_damage` is most likely going to cause a syntax error, see relevant DDA change here: https://github.com/CleverRaven/Cataclysm-DDA/pull/64386

File: data/mods/Magiclysm/items/enchanted_melee.json [hunk 2]
```diff
@@ -755,6 +771,22 @@
...
+    "relative": { "to_hit": 1 }
+  },
+  {
+    "type": "GENERIC",
+    "id": "lucerne_plus_two",
+    "copy-from": "lucern_hammer",
+    "name": { "str": "lucerne hammer +2", "str_pl": "lucerne hammers +2" },
+    "proportional": { "price": 6.0, "price_postapoc": 6.0, "weight": 0.8, "melee_damage": { "all": 1.2 } },
```
Comment (chaosvolt) [inline data/mods/Magiclysm/items/enchanted_melee.json:787]: Same deal here.

File: data/mods/Magiclysm/items/enchanted_melee.json (hunk 1 above)
Comment (EigentlicherVogel) [reply data/mods/Magiclysm/items/enchanted_melee.json:779]: Huh. I swear I had changed all melee damage thing to the bash cut equivalent.

**Review by chaosvolt — COMMENTED** [review]
Loaded up a test build and found no errors or unvisited JSON, so think we're good to go.

EDIT: Oops, thought I'd left it as an approval review, not comment.

**Review by chaosvolt — APPROVED** [review]

**Comment by alp7292** [comment]
recently switched to BN good to see my contribution is in this version too

**Pull request merged**